src/datetime/mod.rs

```diff
@@ -416,6 +416,21 @@ impl<Tz: TimeZone> DateTime<Tz> {
         tz.from_utc_datetime(&self.datetime)
     }
 
+    /// Changes the associated time zone.
+    ///
+    /// The returned `DateTime` references the same instant of time from the perspective of the
+    /// provided time zone.
+    ///
+    /// # Errors
+    ///
+    /// In theory every instant of time should exist in every time zone. Still errors may originate
+    /// from for example an OS API, in which case this method returns `None`.
+    #[inline]
+    #[must_use]
+    pub fn with_timezone_opt<Tz2: TimeZone>(&self, tz: &Tz2) -> Option<DateTime<Tz2>> {
```

Comment:
Something I've been thinking about: for newly fallible API, shouldn't we directly start to yield some kind of `Result` instead of sticking with `Option` everywhere?

Reply:
I sometimes think the same. On the other hand it would make the API seem even less consistent if we end up with a handful of methods that return `Result` and the rest returns `Option`.